### PR TITLE
[PIR] No.46 Migrate paddle.nn.functional.pad into pir

### PR DESCRIPTION
**Pull request by BeingGod**
### PR types

Others

### PR changes

APIs

### Description

Migrate `paddle.nn.functional.pad` into pir
https://github.com/PaddlePaddle/Paddle/issues/57097

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -18,7 +18,12 @@
 from paddle import _C_ops
 from paddle.base.layer_helper import LayerHelper
 from paddle.common_ops_import import Variable, default_main_program
-from paddle.framework import core, in_dynamic_mode, in_pir_mode
+from paddle.framework import (
+    core,
+    in_dynamic_mode,
+    in_dynamic_or_new_ir_mode,
```
Comment (0x45f) [inline python/paddle/nn/functional/common.py]: 函数名已经改为in_dynamic_or_pir_mode

File: python/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -1655,7 +1660,7 @@ def pad(x, pad, mode='constant', value=0.0, data_format="NCHW", name=None):
         paddings = pad
         pad_value = value
 
-        if in_dynamic_mode():
+        if in_dynamic_or_new_ir_mode():
```
Comment (0x45f) [inline python/paddle/nn/functional/common.py]: 同上

**Comment by 0x45f** [comment]
修复了单测中反向的问题

**Review by 0x45f — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**